File: broker/src/main/scala/io/octopus/scala/factory/FixThreadFactory.scala
package io.octopus.scala.factory

import io.netty.util.concurrent.{DefaultThreadFactory, FastThreadLocalThread}

import java.util.concurrent.atomic.AtomicInteger

/**
 * @author chenxu
 * @version 1
 */

class FixThreadFactory(poolName: String) extends DefaultThreadFactory(poolName: String) {

  private val nextId = new AtomicInteger(0)

  override def newThread(r: Runnable): Thread = {
    val t = this.newThread(FastThreadLocalRunnable.wrap(r), this.poolName + "-" + this.nextId.incrementAndGet)
    t.setDaemon(false)
    t.setPriority(5)
    t
  }

  override protected def newThread(r: Runnable, name: String) = new FastThreadLocalThread(this.threadGroup, r, name)

}

File: broker-plugs/broker-mqtt/src/main/java/io/octopus/factory/FixThreadFactory.java
//package io.octopus.factory;
//
//import io.netty.util.concurrent.DefaultThreadFactory;
//import io.netty.util.concurrent.FastThreadLocalThread;
//
//import java.util.concurrent.atomic.AtomicInteger;
//
//
///**
// * @author chenxu
// * @version 1
// */
//public class FixThreadFactory extends DefaultThreadFactory {
//
//    private final String poolName;
//    private final AtomicInteger nextId = new AtomicInteger(0);
//
//    public FixThreadFactory(String poolName) {
//        super(poolName);
//        this.poolName = poolName;
//    }
//
//    @Override
//    public Thread newThread(Runnable r) {
//        Thread t = this.newThread(FastThreadLocalRunnable.wrap(r), this.poolName +"-"+ this.nextId.incrementAndGet());
//        t.setDaemon(false);
//        t.setPriority(5);
//        return t;
//    }
//
//    protected Thread newThread(Runnable r, String name) {
//        return new FastThreadLocalThread(this.threadGroup, r, name);
//    }
//}
